### PR TITLE
[gfx94x] Enable and enhance gfx94x nodes in CI

### PR DESCRIPTION
**Pull request by cderb**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -477,6 +477,10 @@ pipeline {
             name: "TARGET_GFX94X",
             defaultValue: true,
             description: "")
+        booleanParam(
+            name: "TARGET_GFX942",
```
Comment (atamazov) [inline Jenkinsfile]: But this target should be covered by the existing `TARGET_GFX94X` parameter, right?

Comment (cderb) [reply Jenkinsfile]: That's right, we can do without this variable.

**Comment by cderb** [comment]
Shook out errors in the CI machine.
Limited enablement of MI300 CI to dbsync test due to limited hardware.

File: Jenkinsfile [hunk 2]
```diff
@@ -997,6 +1001,25 @@ pipeline {
                                                         needs_gpu:false, needs_reboot:false, build_install: "true")
                     }
                 }
+                stage('dbsync gfx942') {
```
Comment (alexandraBara) [inline Jenkinsfile]:
```suggestion
                stage('Dbsync gfx94X') {
```
or something similar where the first letter is upper case to maintain consistency.

**Review by alexandraBara — APPROVED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -479,6 +479,10 @@ pipeline {
             name: "TARGET_GFX94X",
             defaultValue: false,
             description: "")
+        booleanParam(
+            name: "TARGET_GFX94X_DBSYNC",
```
Comment (atamazov) [inline Jenkinsfile]: [Recommendation] @cderb AFAICS this is workaround so I recommend:
- moving it to the end of the list of params
- renaming this to `WORKAROUND__TARGET_GFX94X__ALWAYS_ENABLE_EVEN_IF_SHORTAGE_OF_HW`, so it becomes cleaner that:
  - This is W/A for some temporary problem
  - This var enables a subset of tests on gfx94x even if all other tests are disabled for this target due to a lack of HW.

File: Jenkinsfile [hunk 4]
```diff
@@ -692,7 +696,7 @@ pipeline {
                     }
                     agent{ label rocmnode("gfx94X") }
                     steps{
-                        buildHipClangJobAndReboot(build_type: 'debug', make_targets: Smoke_targets)
+                        buildHipClangJobAndReboot(build_type: 'debug', make_targets: Smoke_targets, needs_reboot:false)
```
Comment (atamazov) [inline Jenkinsfile:690]: [Question] @cderb Can you please clarify why reboot is disabled for MI300 here and at some other places?

File: Jenkinsfile [hunk 5]
```diff
@@ -961,7 +965,7 @@ pipeline {
                 Navi21_build_cmd = "LLVM_PATH=/opt/rocm/llvm CTEST_PARALLEL_LEVEL=2 MIOPEN_CONV_PRECISE_ROCBLAS_TIMING=0 MIOPEN_LOG_LEVEL=5 make -j\$(nproc) check"
             }
             parallel{
-                stage('dbsync gfx908') {
+                stage('Dbsync gfx908') {
```
Comment (atamazov) [inline Jenkinsfile:959]: [R] We need to add "Dbsync" to the list of available TestSets at line 417. We should have done this a while ago, in PR #2407

File: Jenkinsfile (hunk 3 above)
Comment (atamazov) [inline Jenkinsfile]: I highly recommend enabling also `'Fp32 Hip Debug gfx94X'` stage even is there is a lack of HW.

**Review by atamazov — CHANGES_REQUESTED** [review]
[Recommendation] @cderb I would also appreciate if we resolve 
- https://github.com/ROCm/MIOpen/pull/2407#pullrequestreview-1720670375
- https://github.com/ROCm/MIOpen/pull/2407#pullrequestreview-1792980066

File: Jenkinsfile (hunk 3 above)
Comment (atamazov) [reply Jenkinsfile]: btw this justifies renaming requested at https://github.com/ROCm/MIOpen/pull/2518#discussion_r1567442010

**Comment by junliume** [comment]
@cderb could you follow up on this PR?

File: Jenkinsfile (hunk 4 above)
Comment (cderb) [reply Jenkinsfile:690]: This is to avoid disabling the machine, as sometimes the MI300 node isn't reconnecting after reboot.

**Comment by cderb** [comment]
Resolved test node directory permission issue.

**Comment by cderb** [comment]
@atamazov @junliume this is ready.

**Review by junliume — APPROVED** [review]

**Pull request merged**